core/tee/tee_svc_storage.c

```diff
@@ -992,12 +1030,16 @@ TEE_Result tee_svc_storage_start_enum(uint32_t obj_enum, uint32_t storage_id)
 	/* re-start */
 	res = tee_file_ops.closedir(e->dir);
 	e->dir = NULL;
-	if (res != 0)
-		return TEE_ERROR_GENERIC;
+	if (res) {
```

Comment:
Is not it more than refactoring, but also potential memory leak fix? (same below)